samples/nrf9160/modem_shell/README.rst

```diff
@@ -0,0 +1,409 @@
+
+nRF9160: Modem Shell
+####################
+
+The Modem Shell (MoSh) sample application enables you to test various device connectivity features, including data throughput.
+
+Overview
+********
+
+MoSh enables testing of different connectivity features such as
+LTE link handling, TCP/IP connections, data throughput (iperf3 and curl), SMS, GNSS, FOTA updates and PPP.
+Hence, this sample is not only code sample, which it also is in many aspects, but also a test application for aforementioned features.
+MoSh uses the LTE link control driver to establish an LTE connection and
+initializes the Zephyr shell to provide a shell command-line interface for users.
+
+Below subsections list MoSh features and show shell command examples on how they can be used.
+
+LTE link control
+================
+
+MoSh command: "link"
+
+LTE link control is used to change and query the state of the LTE connection.
+Many of the changes are applied when going to online mode for the next time.
+
+Examples
+--------
+
+Change system mode to NB-IoT and query connection status::
+
+   link funmode -4
+   link sysmode -n
+   link funmode -1
+   link status
+
+Open another PDP context, and close it::
+
+   link connect -a internet.operator.com
+   link status
+   link disconnect -I 1
+
+Enable and configure eDRX::
+
+   link edrx -e -m --edrx_value 0010 --ptw 0010
+
+Enable and configure PSM::
+
+   link psm -e --rptau 01100100 --rat 00100010
+
+AT commands
+===========
+
+MoSh command: "at"
+
+AT command module can be used to send AT commands to the modem.
+
+Examples
+--------
+
+Send AT command to query network status with::
+
+   at at+cereg?
+
+Send AT command to query neightbour cells::
+
+   at at%NBRGRSRP
+
+Enable AT command events::
+
+   at events_enable
+
+Disable AT command events::
+
+   at events_disable
+
+Ping
+====
+
+MoSh command: "ping"
+
+Ping is a well-known tool for testing the reachability of a host on an IP network.
+
+Examples
+--------
+
+Ping a URL::
+
+   ping -d ping.server.url
+
+Ping an IPv6 address with length of 500 bytes and 1s intervals::
+
+   ping -d 1a2b:1a2b:1a2b:1a2b::1 -6 -l 500 -i 1000
+
+Iperf3
+======
+
+MoSh command: "iperf3"
+
+Iperf3 is a well-known tool for measuring data transfer performance both in uplink and downlink direction.
+
+Examples
+--------
+
+Download data for 30 seconds with packet size of 3540 bytes and use detailed output::
+
+   iperf3 --client 111.222.111.222 --port 10000 -l 3540 --time 30 -V -R
+
+Upload data for 60 seconds with packet size of 3540 bytes and use detailed output as well as debug output::
+
+   iperf3 --client 111.222.111.222 --port 10000 -l 3540 --time 60 -V -d
+
+Download data for 10 seconds over IPv6::
+
+   iperf3 --client 1a2b:1a2b:1a2b:1a2b::1 --port 20000 --time 10 -R -6
+
+Curl
+====
+
+MoSh command: "curl"
+
+Curl is a well-known tool for doing all sorts of URL manipulations and transfers.
+It's part of MoSh so that testing data download can be performed with a "standard" tool.
```

Comment:
```suggestion
It is a part of MoSh and enables you to download testing data with a "standard" tool.
```

Reply:
I think the meaning changes here from `testing data download` to `download testing data`, which sounds different to me.

Reply:
You are so right. Check now if I got it correct.

Reply:
This is good.